homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -147,84 +146,81 @@ def __init__(
         self._timezone = get_time_zone("Europe/Stockholm")
+        self._attr_extra_state_attributes = {
+            ATTR_DEPARTURE_STATE: None,
+            ATTR_CANCELED: None,
+            ATTR_DELAY_TIME: None,
+            ATTR_PLANNED_TIME: None,
+            ATTR_ESTIMATED_TIME: None,
+            ATTR_ACTUAL_TIME: None,
+            ATTR_OTHER_INFORMATION: None,
+            ATTR_DEVIATIONS: None,
+        }
 
     async def async_update(self) -> None:
         """Retrieve latest state."""
+        when = datetime.now()
         if self._time:
             departure_day = next_departuredate(self._weekday)
             when = datetime.combine(departure_day, self._time).astimezone(
                 self._timezone
             )
-            try:
+        try:
+            if self._time:
                 self._state = await self._train_api.async_get_train_stop(
                     self._from_station, self._to_station, when
                 )
-            except ValueError as output_error:
-                _LOGGER.error(
-                    "Departure %s encountered a problem: %s", when, output_error
-                )
-        else:
-            when = datetime.now()
-            try:
+            else:
                 self._state = await self._train_api.async_get_next_train_stop(
                     self._from_station, self._to_station, when
                 )
-            except ValueError as output_error:
-                _LOGGER.error(
-                    "Departure %s encountered a problem: %s", when, output_error
-                )
+        except ValueError as output_error:
+            _LOGGER.error("Departure %s encountered a problem: %s", when, output_error)
+
         if self._state:
-            self._departure_state = self._state.get_state().name
-            self._delay_in_minutes = self._state.get_delay_time()
 
-    @property
-    def extra_state_attributes(self) -> dict[str, Any] | None:
-        """Return the state attributes."""
-        if not self._state:
-            return None
-        attributes = {
-            ATTR_DEPARTURE_STATE: self._departure_state,
-            ATTR_CANCELED: self._state.canceled,
-            ATTR_DELAY_TIME: None,
-            ATTR_PLANNED_TIME: None,
-            ATTR_ESTIMATED_TIME: None,
-            ATTR_ACTUAL_TIME: None,
-            ATTR_OTHER_INFORMATION: None,
-            ATTR_DEVIATIONS: None,
-        }
-        if self._state.other_information:
-            attributes[ATTR_OTHER_INFORMATION] = ", ".join(
-                self._state.other_information
-            )
-        if self._state.deviations:
-            attributes[ATTR_DEVIATIONS] = ", ".join(self._state.deviations)
-        if self._delay_in_minutes:
-            attributes[ATTR_DELAY_TIME] = self._delay_in_minutes.total_seconds() / 60
-        if self._state.advertised_time_at_location:
-            attributes[ATTR_PLANNED_TIME] = as_utc(
+            self._attr_native_value = (
```

Comment:
If you invert the logic and exit if not self._state, then you can un-indent all that follows.
eg.
```python
        if not self._state:
              self._attr_available = False
              self._attr_native_value = None
              self._attr_extra_state_attributes = None
              return
```